addons/base_iban/models/res_partner_bank.py

```diff
@@ -133,12 +155,18 @@ def _check_iban(self):
             if bank.acc_type == 'iban':
                 validate_iban(bank.acc_number)
 
-    def check_iban(self, iban=''):
-        try:
-            validate_iban(iban)
-            return True
-        except ValidationError:
-            return False
+    @api.model
+    def get_acc_number_type(self, acc_number):
```

Comment:
This should be refactored. Only this function is used in RPC calls.

You have a selection field with a compute, `get_acc_number_type` and `retrieve_acc_type`. You could have just one function in base module that returns a tuple (code, label) and the compute based on it.